### PR TITLE
Improve Canberra metric

### LINKED ISSUE #1311: Improve Canberra metric
## 🚀 Feature

Actual implementation of Canberra metric does not use absolute value on terms in denominator. Moreover, `sklearn` can be used in test.

See https://arxiv.org/pdf/1411.7474.pdf 

See https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.DistanceMetric.html

### PR DESCRIPTION
**Pull request by sdesrozis**
Fixes #1311 

Description:

Fix Canberra to use abs in denominator. Use `sklearn` in test.

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [x] Documentation is updated (if required)

### CODE REVIEW

File: ignite/contrib/metrics/regression/canberra_metric.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def reset(self):
 
     def _update(self, output):
         y_pred, y = output
-        errors = torch.abs(y.view_as(y_pred) - y_pred) / (y_pred + y.view_as(y_pred))
+        errors = torch.abs(y.view_as(y_pred) - y_pred) / (torch.abs(y_pred) + torch.abs(y.view_as(y_pred)))
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/canberra_metric.py:27]: Please, update the docstring formula and also add a link to sklearn.

**Review by vfdev-5 — COMMENTED** [review]
Thanks @sdesrozis , looks good !

File: tests/ignite/contrib/metrics/regression/test_canberra_metric.py [hunk 2]
```diff
@@ -30,18 +31,27 @@ def test_compute():
...
+    np_sum += value
     assert m.compute() == pytest.approx(np_sum)
+    assert canberra.pairwise([c, ground_truth])[0][1] == pytest.approx(value)
 
     m.update((torch.from_numpy(d), torch.from_numpy(ground_truth)))
-    np_sum += (np.abs(ground_truth - d) / (d + ground_truth)).sum()
+    value = (np.abs(ground_truth - d) / (np.abs(d) + np.abs(ground_truth))).sum()
+    np_sum += value
     assert m.compute() == pytest.approx(np_sum)
+    assert canberra.pairwise([d, ground_truth])[0][1] == pytest.approx(value)
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/regression/test_canberra_metric.py:57]: Can you also test the total computed metric with sklearn and not a single update ? Same for above.

Comment (sdesrozis) [reply tests/ignite/contrib/metrics/regression/test_canberra_metric.py]: @vfdev-5 done

File: ignite/contrib/metrics/regression/canberra_metric.py (hunk 1 above)
Comment (sdesrozis) [reply ignite/contrib/metrics/regression/canberra_metric.py:27]: @vfdev-5 done

Comment (vfdev-5) [reply ignite/contrib/metrics/regression/canberra_metric.py:27]: How about the link to sklearn ?

**Review by vfdev-5 — APPROVED** [review]
👍
